src/renderer/views/deposit/DepositView.tsx

```diff
@@ -149,42 +158,26 @@ export const DepositView: React.FC<Props> = () => {
 
   const poolDetailRD = useObservableState<PoolDetailRD>(selectedPoolDetail$, RD.initial)
 
-  const [liquidityProvider] = useObservableState<LiquidityProviderRD>(
-    () =>
-      FP.pipe(
-        Rx.combineLatest([
-          network$,
-          // We should look for THORChain's wallet at the response of liqudity_providers endpoint
-          addressByChain$(THORChain),
-          address$,
-          assetWithDecimalLD
-        ]),
-        RxOp.switchMap(([network, oAddress, oAssetAddress, assetWithDecimalRD]) =>
-          FP.pipe(
-            sequenceTOption(RD.toOption(assetWithDecimalRD)),
-            O.fold(
-              (): LiquidityProviderLD => Rx.EMPTY,
-              ([assetWithDecimal]) =>
-                getLiquidityProvider({ assetWithDecimal, network, runeAddress: oAddress, assetAddress: oAssetAddress })
-            )
+  const [liquidityProvider] = useObservableState<LiquidityProviderRD>(() => {
+    return Rx.combineLatest([
+      network$,
+      // We should look for THORChain's wallet at the response of liqudity_providers endpoint
+      address$,
+      addressByChain$(THORChain),
+      assetWithDecimalLD
+    ]).pipe(
+      RxOp.switchMap(([network, oAssetAddress, oRuneAddress, assetWithDecimalRD]) => {
+        return FP.pipe(
+          sequenceTOption(oRuneAddress, oAssetAddress, RD.toOption(assetWithDecimalRD)),
+          O.fold(
+            (): LiquidityProviderLD => Rx.of(RD.initial),
+            ([runeAddress, assetAddress, assetWithDecimal]) =>
+              getLiquidityProvider({ assetWithDecimal, network, runeAddress, assetAddress })
```

Comment:
i would move this data getter directly inside of `SymDepositView`. with optional `runeAddress` at the `getLiquidityProvider` we can easily split this logic between 2 views without passing-down `liquidityProvider` through few levels:
- `SymDepositView`: `getLiquidityProvider({ assetWithDecimal: ..., network: ..., runeAddress: O.some(Address), assetAddress: Address })`
- `AsymDepositView`: `getLiquidityProvider({ assetWithDecimal: ..., network: ..., runeAddress: O.none, assetAddress: Address })`

what do you think ?

Reply:
Sounds good.

Reply:
_Will be done with another PR - see #1512_

Reply:
I'm going to skip this for same reason mentioned in https://github.com/thorchain/asgardex-electron/pull/1505#discussion_r647415243